File: DSA/LakshaKundra_590013237/Day_71_Question_1.py
class Node:
    def __init__(self, key):
        self.left = None
        self.right = None
        self.val = key

def inorder(root):
    if root:
        inorder(root.left)
        print(root.val, end=" ")
        inorder(root.right)

root1 = Node(1)
root1.right = Node(3)
root1.right.left = Node(2)
inorder(root1)
print()

root2 = Node(4)
root2.left = Node(2)
root2.right = Node(5)
inorder(root2)
print()

File: DSA/PriyankaSingh_590016827/day71question1.c
#include <stdio.h>
#include <stdlib.h>

// Definition for a binary tree node.
typedef struct TreeNode {
    int data;
    struct TreeNode* left;
    struct TreeNode* right;
} TreeNode;

// Function to create a new tree node
TreeNode* createNode(int data) {
    TreeNode* newNode = (TreeNode*)malloc(sizeof(TreeNode));
    newNode->data = data;
    newNode->left = NULL;
    newNode->right = NULL;
    return newNode;
}

// Inorder Traversal of Binary Tree
void inorderTraversal(TreeNode* root) {
    if (root == NULL) {
        return;
    }
    inorderTraversal(root->left);
    printf("%d ", root->data);
    inorderTraversal(root->right);
}

int main() {
    // Example 1
    TreeNode* root1 = createNode(1);
    root1->right = createNode(3);
    root1->right->left = createNode(2);
    printf("Inorder Traversal: ");
    inorderTraversal(root1);
    printf("\n");

    // Example 2
    TreeNode* root2 = createNode(4);
    root2->left = createNode(2);
    root2->right = createNode(5);
    printf("Inorder Traversal: ");
    inorderTraversal(root2);
    printf("\n");

    return 0;
}
